### DISCUSSION TITLE
debug_console输出一个服务的task非常多，这个是代表了什么

### DISCUSSION THREAD

**Discussion by ericzhanchina** [cloudwu/skynet#1728] [General]
今天游戏刚刚小范围玩家测试，300人同时在线，我发现整个gated服，有这么多的task，这个参数是代表了什么，是我的代码写的有问题造成的吗？

0000000e	cpu:18.77872	message:463789	mqlen:0	task:3153

:0000000e	snlua gated login logind

**Comment by hanxi** [comment]
https://github.com/cloudwu/skynet/blob/master/lualib/skynet.lua#L971-L977

session_id_coroutine 的数据量

**Reply by ericzhanchina** [reply]
您好，我看了一下代码，是不是任何没有完成的协程都会在这里挂起，感觉我只有这个gate服务有这么多task是不正常的，这个gate主要是给msgagent分发消息的，与login服连接管理用户登录，我如何能分析是哪些任务挂起的呢？

**Comment by ericzhanchina** [comment]
看到可以task address，我先看看调用栈
